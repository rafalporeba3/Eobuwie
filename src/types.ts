import { Ref } from '@vue/composition-api';

export type Computed<T> = Readonly<Ref<Readonly<T>>> | Ref<Readonly<T>>;
