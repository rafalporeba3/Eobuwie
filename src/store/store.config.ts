import { Module }               from 'vuex';
import { applicationRootState } from '@/store/store.state';
import { mutations }            from '@/store/mutations/store.mutations';
import { ApplicationRootState } from '@/store/store.types';

const storeConfig: Module<ApplicationRootState, {}> = {
  state: applicationRootState,
  mutations,
  getters: {},
  actions: {},
};

export default storeConfig;
