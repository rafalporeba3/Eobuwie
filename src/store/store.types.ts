export interface RootState {
  [ key: string ]: any;
}

export interface ApplicationRootState {
  currency: string;
}
