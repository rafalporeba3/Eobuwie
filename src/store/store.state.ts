import { ApplicationRootState } from '@/store/store.types';

export const applicationRootState: ApplicationRootState = {
  currency: 'PLN',
};
