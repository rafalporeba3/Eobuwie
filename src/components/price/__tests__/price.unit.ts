import Vuex, { Store }               from 'vuex';
import { createLocalVue, Wrapper }   from '@vue/test-utils';
import CompositionApi                from '@vue/composition-api';
import { createTestInstanceShallow } from '@/__tests__/factory';
import { RootState }                 from '@/store/store.types';
import cPrice                        from '../price.component.vue';

const localVue = createLocalVue();
localVue.use(CompositionApi);
localVue.use(Vuex);

describe('Price component', () => {
  let wrapper!: Wrapper<any>;
  let store!: Store<RootState>;

  beforeEach(() => {
    const {
      wrapperInstance,
      storeInstance,
    } = createTestInstanceShallow(cPrice, {
      localVue,
      propsData: {
        price: 300,
      },
    });
    wrapper = wrapperInstance;
    store = storeInstance;
  });

  it('should render cReservationForm component with proper class', () => {
    const priceComponent: Wrapper<any> = wrapper.findComponent(cPrice);

    expect(priceComponent.exists())
      .toBeTruthy();

    expect(priceComponent.classes())
      .toContain('c-price');
  });

  it('should contain value and currency elements', () => {
    const valueElement = wrapper.find('.c-price__value');
    const currencyElement = wrapper.find('.c-price__currency');

    expect(valueElement.exists())
      .toBeTruthy();
    expect(currencyElement.exists())
      .toBeTruthy();
  });

  it('should contain correct value', () => {
    expect(wrapper.props('price'))
      .toBe(300);
  });

  it('should contain correct currency', () => {
    const currencyElement = wrapper.find('.c-price__currency');

    expect(currencyElement.text())
      .toBe('zÅ‚');
  });

  it('should match snapshot', () => {
    expect(wrapper.html())
      .toMatchSnapshot();
  });
});
