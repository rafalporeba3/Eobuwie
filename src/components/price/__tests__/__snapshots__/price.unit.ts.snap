// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component should match snapshot 1`] = `<div class="c-price"><span class="c-price__value">300</span> <span class="c-price__currency">z≈Ç</span></div>`;
